homeassistant/components/history_stats/data.py

```diff
@@ -88,20 +88,20 @@ async def async_update(
         if current_period_start_timestamp > now_timestamp:
             # History cannot tell the future
             self._history_current_period = []
-            self._previous_run_before_start = True
+            self._previous_run_before_start = False
```

Comment:
I spent a while looking at this variable and I think it is just implemented backwards from how it is described. 

It seems to be checking if we have done a previous run, but every time we do a run we set it to False. The only time it is set true is in this bogus case where we query the future.

I flipped it such that:
 - initializes to false
 - sets to false if we hit this future query case
 - sets to true when we do a successful query

Reply:
Should we rename it?

Reply:
I would almost say I'd like to remove it entirely, as I think you could do without it, but that adds maybe a bit too much refactoring to this PR?

For now maybe I will just call it `_has_recorder_data` as I think that's what it's really saying.

Reply:
> I would almost say I'd like to remove it entirely, as I think you could do without it, but that adds maybe a bit too much refactoring to this PR?

Yeah, probably best to rename and not remove to keep this from getting larger

`_has_recorder_data` 👍